### PR TITLE
Change truncation from 3 to 5 characters

### PR DESCRIPTION
**Pull request by jpmckinney**
As mentioned in #118, these paths currently collide on `con_imp_tra_pay_additionalIdent`

* `contracts/implementation/transactions/payee/additionalIdentifiers`
* `contracts/implementation/transactions/payer/additionalIdentifiers`

I.e. `payee` and `payer` have the same three first letters.

I changed the prefix length to 5 characters.

With three characters, you can theoretically support seven parents as long as no properties at any level start with the same three letters (`aaa_bbb_ccc_ddd_eee_fff_ggg_xyz`).

With five characters, you can theoretically support four parents, with the same caveat (`aaaaa_bbbbb_ccccc_ddddd_abcdefg`).

### CODE REVIEW

**Comment by jpmckinney** [comment]
Let's default to 5 (to avoid accidental collision with OCDS data). We can open a new issue to make this configurable.

**Comment by odscjames** [comment]
I'm not happy about changing the default. This tool is used for different standards and for different purposes, and for there to suddenly be a change just to suit one purpose doesn't seem great. I would strongly prefer to leave the default at 3, just for historical reasons.

That said, I'm totally happy to have a new config option that can be set for OCDS or in CoVE OCDS. I don't think that would be hard to add - I'm happy to have a go when I have time (not sure when).

**Comment by jpmckinney** [comment]
I've left the default at 3, and made it configurable. Please create an issue to use 5 in CoVE OCDS.

File: flattentool/cli.py [hunk 1]
```diff
@@ -180,6 +188,10 @@ def create_parser():
         "--convert-titles",
         action='store_true',
         help="Convert titles. Requires a schema to be specified.")
+    parser_unflatten.add_argument(
```
Comment (kindly) [inline flattentool/cli.py]: Should not be relevant to unflatten as it uses sub sheet names from the actual spreadheet.

**Comment by jpmckinney** [comment]
@kindly I've removed those occurrences. I wasn't sure if perhaps unflatten, when provided a schema, calculated sub-sheet names, to determine a correspondence between the flat data (which might have been created with a non-default truncation length) and the schema.

**Comment by kindly** [comment]
@jpmckinney FYI I think the design was that sheet names (apart from Meta) have no relevance at all when unflattening.

**Comment by odscjames** [comment]
Hello - just to note this has no change log entry, and no new docs. The former is really required for any work and the latter is always good, especially for new stuff like this. I have started this on #281 - hopefully we can fix this and merge in ASAP. Thanks,

(I'll make a OCDS CoVE issue now.)

**Pull request merged**